### PR TITLE
Add support for horsey theme

### LINKED ISSUE #5: Improve horsey theme with images overlay
For now when `horsey` background is selected, standard move, selected piece, etc. solid color overlays are applied. 

`horsey` them need special image overlays instead. Those images can be found here: https://github.com/lichess-org/lila/tree/master/public/images/board

### PR DESCRIPTION
**Pull request by csdodd**
Fixes #5 

The `horsey` theme uses images for square highlights as oppose to solid colors. This change introduces a new structure called HighlightDetails that allows for a theme to specify if it wishes to highlight using a solid color or an image.

When rendering, if an image is present, it will take precedent over a solid color.

Screenshot_20230509_101440

### CODE REVIEW

**Comment by csdodd** [comment]
There is one bug that this introduces.

GIVEN the user is using the `horsey` theme
AND the opponent has moved a piece that can be taken
WHEN the user selects the piece that will take their opponents piece
THEN the last move background image should not be visible


Here is a screenshot of the bug -- the image background should be removed when the piece can be taken.

Screenshot_20230509_102009

It's a little tricky to fix this since there a whole render pipeline inside a stack so messing with the logic needs to be done carefully -- so I have opted to not fix this issue in this PR.

**Comment by veloce** [comment]
Not necessarily a bug I would say.

File: lib/src/board_color_scheme.dart [hunk 1]
```diff
@@ -654,9 +654,19 @@ class BoardColorScheme {
...
     validMoves: Color(0x4014551e),
     validPremoves: Color(0x40203085),
   );
 }
+
+class HighlightDetails {
+  const HighlightDetails({
+    this.solidColor,
+    this.image,
+  });
```
Comment (veloce) [inline lib/src/board_color_scheme.dart:668]: Here you could add an `assert` to ensure we don't forget to provide at least one of the arg.
Sth like:
```dart
 const HighlightDetails({
     this.solidColor,
     this.image,
   }) : assert(solidColor != null || image != null, 'You must provider either `solidColor` or `image`.');
```

**Comment by csdodd** [comment]
> I think it is currently the same result whether you choose horsey theme or not, isn't it?

Good point -- just checked and you are correct. This bug is present in all themes.

**Pull request merged**